pkg/module/go_get_fetcher.go

```diff
@@ -0,0 +1,156 @@
+	repoURI string
+	version string
+}
+
+// NewGoGetFetcher creates fetcher which uses go get tool to fetch modules
+func NewGoGetFetcher(fs afero.Fs, repoURI, version string) (Fetcher, error) {
+	if repoURI == "" {
+		return nil, errors.New("invalid repository identifier")
+	}
+
+	return &goGetFetcher{
+		fs:      fs,
+		repoURI: repoURI,
+		version: version,
+	}, nil
+}
+
+// Fetch downloads the sources and returns path where it can be found. Fetch will always
+// return a non-nil Ref.
+//
+// TODO: If an error is returned, the Fetch method will likely fail, but you should always call
+// Close on the returned
+func (g *goGetFetcher) Fetch(mod, ver string) (Ref, error) {
+	// setup the GOPATH
+	goPathRoot, err := afero.TempDir(g.fs, "", "athens")
+	if err != nil {
+		// TODO: return a ref for cleaning up the goPathRoot
+		return nil, err
+	}
+	sourcePath := filepath.Join(goPathRoot, "src", goPathRoot)
+	modPath := filepath.Join(sourcePath, getRepoDirName(g.repoURI, g.version))
+	if err := g.fs.MkdirAll(modPath, os.ModeDir|os.ModePerm); err != nil {
+		// TODO: return a ref for cleaning up the goPathRoot
+		return nil, err
+	}
+
+	// setup the module with barebones stuff
+	if err := prepareStructure(g.fs, modPath); err != nil {
+		// TODO: return a ref for cleaning up the goPathRoot
+		return nil, err
+	}
+
+	cachePath, err := getSources(g.fs, goPathRoot, modPath, mod, ver)
+	if err != nil {
+		// TODO: return a ref that cleans up the goPathRoot
+		return newDiskRef(g.fs, cachePath, ver), err
+	}
+	// TODO: make sure this ref also cleans up the goPathRoot
+	diskRef := newDiskRef(g.fs, cachePath, ver)
+
+	return diskRef, err
+}
+
+// Hacky thing makes vgo not to complain
+func prepareStructure(fs afero.Fs, repoRoot string) error {
+	// vgo expects go.mod file present with module statement or .go file with import comment
+	gomodPath := filepath.Join(repoRoot, "go.mod")
+	gomodContent := []byte("module \"mod\"")
+	if err := afero.WriteFile(fs, gomodPath, gomodContent, 0666); err != nil {
+		return err
+	}
+
+	sourcePath := filepath.Join(repoRoot, "mod.go")
+	sourceContent := []byte(`package mod // import "mod"`)
+	return afero.WriteFile(fs, sourcePath, sourceContent, 0666)
+}
+
+// given a filesystem, gopath, repository root, module and version, runs 'vgo get'
+// on module@version from the repoRoot with GOPATH=gopath, and returns the location
+// of the module cache. returns a non-nil error if anything went wrong. always returns
+// the location of the module cache so you can delete it if necessary
+func getSources(fs afero.Fs, gopath, repoRoot, module, version string) (string, error) {
+	version = strings.TrimPrefix(version, "@")
+	if !strings.HasPrefix(version, "v") {
+		version = "v" + version
+	}
+	uri := strings.TrimSuffix(module, "/")
+
+	fullURI := fmt.Sprintf("%s@%s", uri, version)
+
+	gopathEnv := fmt.Sprintf("GOPATH=%s", gopath)
+	cacheEnv := fmt.Sprintf("GOCACHE=%s", filepath.Join(gopath, "cache"))
+	disableCgo := "CGO_ENABLED=0"
```

Comment:
I copied this code from `./pkg/repo` so I'm not entirely familiar, but my guess would be so that we can run this code inside of non-libc runtimes, like an `alpine` docker container